ports/geckolib/wrapper.rs

```diff
@@ -108,13 +107,9 @@ impl TRestyleDamage for GeckoRestyleDamage {
 
     fn compute(source: &nsStyleContext,
                new_style: &Arc<ComputedValues>) -> Self {
-        type Helpers = ArcHelpers<ServoComputedValues, ComputedValues>;
         let context = source as *const nsStyleContext as *mut nsStyleContext;
-
-        Helpers::borrow(new_style, |new_style| {
-            let hint = unsafe { Gecko_CalcStyleDifference(context, new_style) };
-            GeckoRestyleDamage(hint)
-        })
+        let hint = unsafe { Gecko_CalcStyleDifference(context, ComputedValues::to_borrowed(new_style)) };
```

Comment:
@Manishearth and I discussed about this, and I *think* the conclusion is that coherency (read with programming-language jargon) does not block us in this case.

TL;DR; what I'm proposing is something like [this](https://play.rust-lang.org/?gist=3985f917ee34f52b29e9d27004213007&version=stable&backtrace=0).

Discussion: http://logs.glob.uno/?c=mozilla%23servo&s=17+Aug+2016&e=17+Aug+2016#c504578